pysingfel/experiment/holo.py

```diff
@@ -70,7 +70,7 @@ def get_ref_position(self):
         Return the position.
         """
         if self._ref_position is None:
-            return None
+            return np.array([[0.,0.,0.]])
```

Comment:
I had a discussion with Iris about this recently. 
I think if a variable is None, it should just return None.

Reply:
Would it make sense then to have the default variable value be `np.array([[0.,0.,0.]])` rather than `None` in setting up the class? I changed this from None because it causes the `generate_new_sample_state` function to throw an error from trying to concatenate NoneType and an array. I saw that in a previous version of the code, `np.array([[0., -1e-6, 0.]])` was returned (instead of all zeros); is there a preferred default position for the reference particle if the user doesn't specify one?

Reply:
@apeck12 How about if we add a check for positions/orientations = "None" in `generate_new_sample_state` here:
https://github.com/chuckie82/pysingfel/blob/1985f5cbc13ce5be54e1ee0ffaa4303088b51181/pysingfel/experiment/holo.py#L49

@irischang020 Having revisited the code, I think the user should set these arguments at initialization. There's no good default values for these. Also, please add description in the docstring for how to set these values, what datatype, positions respect to what? etc.
https://github.com/chuckie82/pysingfel/blob/1985f5cbc13ce5be54e1ee0ffaa4303088b51181/pysingfel/experiment/holo.py#L16

ExampleHolography.py should also show an example of how to set these values.

Reply:
I’ve added a line to check that neither the reference particle orientation or position is None in the latest commit.

Reply:
We should check part_positions and part_orientations are not None as well
If None, should we print an error and exit?

And return None in `get_ref_position`

Reply:
Sorry, I’m not sure I follow the second comment about return None in `get_ref_position`. For `part_positions` and `part_orientations`, these sample particle variables are currently handled in a way to ensure they aren’t None before simulating each shot since the function `generate_new_sample_state` only calls the getter functions (which internally override the variable values if None) and not the setter functions.

Reply:
could you point me to the line of code that ensures `part_positions` is not None?

Reply:
For each simulated shot, `generate_new_sample_state` is called and in turn calls `get_next_part_position`. If the default value is None (either from initialization or from resetting the value with `set_part_positions`), then an array of zeros is returned: https://github.com/apeck12/pysingfel/blob/b3ebce18f71769ba26e57efdf04c5651ced4b044/pysingfel/experiment/holo.py#L110. 
Similarly, a random quaternion is returned if `part_orientations` is None. The `part_positions` and `part_orientations` variables aren't used outside of `generate_new_sample_state`.

Reply:
ok, thanks. I think returning position [0,0,0] when the value is None isn't a good programming practice.

I'll merge this, but looks like we will need think more about the design of holo.py